### PR TITLE
Refactor dvdz

### PR DESCRIPTION
**Pull request by ntejos**
Idea is to refactor functions related to passing from dv to dz and vice-versa. This PR should address issue #330 . 

Please review this initial commit without merging before starting the major refactor.

### CODE REVIEW

**Comment by ntejos** [comment]
@profxj can you have a look at this first commits before proceeding further?

File: linetools/utils.py [hunk 1]
```diff
@@ -346,84 +352,134 @@ def z_from_v(z, v):
...
         Whether to apply relativistic correction for
         a locally flat space-time. Default is True.
 
     Returns
     -------
     dv : Quantity or Quantity array
-        Rest frame velocity difference between z and zmean, at
-        zmean. It has same shape as z.
+        Rest-frame velocity difference between z and zref (dv=0 at zref by definition).
+        It has same shape as z.
```
Comment (profxj) [inline linetools/utils.py]: the same

File: linetools/utils.py [hunk 2]
```diff
@@ -346,84 +352,134 @@ def z_from_v(z, v):
...
     z : float or np.array
         Redshifts to calculate dv on
-    zmean : float or np.array
-        Rest-frame redshift to perform the calculation.
-        If shape of zmean is equal than shape of z,
-        each dv is calculated at each zmean, otherwise zmean
-        is expected to be float
+    zref : float or np.array
+        Reference redshift where dv=0 by definition.
+        If shape of zref is equal than shape of z,
```
Comment (profxj) [inline linetools/utils.py]: equal to the

File: linetools/utils.py [hunk 3]
```diff
@@ -346,84 +352,134 @@ def z_from_v(z, v):
...
+        each dv is calculated at each zref, otherwise zref
+        es expected to be float.
```
Comment (profxj) [inline linetools/utils.py]: is

File: linetools/utils.py [hunk 4]
```diff
@@ -346,84 +352,134 @@ def z_from_v(z, v):
...
     -------
     dv : Quantity or Quantity array
-        Rest frame velocity difference between z and zmean, at
-        zmean. It has same shape as z.
+        Rest-frame velocity difference between z and zref (dv=0 at zref by definition).
+        It has same shape as z.
         """
     z = np.array(z)
-    zmean = np.array(zmean)
+    zref = np.array(zref)
```
Comment (profxj) [inline linetools/utils.py]: You'll want to add checks as regards the rules you have in the docs
for the shapes of z and zref.

File: linetools/utils.py [hunk 5]
```diff
@@ -346,84 +352,134 @@ def z_from_v(z, v):
...
+    Notes
+    -----
+    See also linetools.utils.z_from_dv()
+    """
     if not isinstance(dv, u.quantity.Quantity):
-        raise ValueError('dv must be Quantity or Quantity array!')
+        raise ValueError('dv must be Quantity or Quantity array.')
     try:
         dv = dv.to('km/s') # dv in km/s
         dv = np.array(dv.value)
```
Comment (profxj) [inline linetools/utils.py]: I'd let astropy units do their thing down below, i.e. don't take the value here

File: linetools/utils.py [hunk 6]
```diff
@@ -346,84 +352,134 @@ def z_from_v(z, v):
...
     except UnitConversionError:
-        raise ValueError('dv must have velocity units!')
+        raise ValueError('dv must have velocity units.')
 
-    zmean = np.array(zmean)
+    zref = np.array(zref)
 
     if rel:
         beta = dv / const.c.to('km/s').value
```
Comment (profxj) [inline linetools/utils.py]: leave units on dv and const.c and decompose beta

File: linetools/utils.py (hunk 2 above)
Comment (ntejos) [inline linetools/utils.py]: Corrected.

File: linetools/utils.py (hunk 3 above)
Comment (ntejos) [inline linetools/utils.py]: Done.

File: linetools/utils.py (hunk 1 above)
Comment (ntejos) [inline linetools/utils.py]: Thanks.

File: linetools/utils.py (hunk 4 above)
Comment (ntejos) [inline linetools/utils.py]: Done.

File: linetools/utils.py (hunk 5 above)
Comment (ntejos) [inline linetools/utils.py]: Done.

File: linetools/utils.py (hunk 6 above)
Comment (ntejos) [inline linetools/utils.py]: Done.

**Comment by ntejos** [comment]
Previous comments addressed, and it seems the refactor has been properly completed at least within linetools. Things may still break outside linetools but they should be easy to fix. This PR addresses issue #330.

**Comment by profxj** [comment]
Looks very good.

**Pull request merged**